### PR TITLE
feat: add etcd store package

### PR DESCRIPTION
**Pull request by starsz**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
```
Comment (tokers) [inline .github/workflows/test.yml:10]: We may add concurrent settings: https://github.com/api7/cloud/blob/main/.github/workflows/e2e-tests.yaml#L10.

File: etcdstore.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	"go.etcd.io/etcd/clientv3"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/securecookie"
+	"github.com/gorilla/sessions"
+)
+
+var (
+	ErrNilClient   = errors.New("etcd client is nil")
```
Comment (tokers) [inline etcdstore.go:17]: May add some explanations for these two errors.

**Comment by tokers** [comment]
@starsz How did you set TTL for these session data?

**Comment by starsz** [comment]
> @starsz How did you set TTL for these session data?

https://github.com/api7/etcdstore/pull/1/files#diff-10b0efd3f5b9f5a91fbd0ca256e6360a255e0c939ea0031df6ebe6fb19f8c353R95-R105

Use lease to set the ttl

**Comment by tokers** [comment]
> > @starsz How did you set TTL for these session data?
> 
> https://github.com/api7/etcdstore/pull/1/files#diff-10b0efd3f5b9f5a91fbd0ca256e6360a255e0c939ea0031df6ebe6fb19f8c353R95-R105
> 
> Use lease to set the ttl

Sorry, I didn't notice it.

**Review by tokers — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (starsz) [reply .github/workflows/test.yml]: Fixed.

File: etcdstore.go (hunk 2 above)
Comment (starsz) [reply etcdstore.go]: These errors are not used. So I delete them.

**Pull request merged**